Сетка страниц и крупных блоков на флексах

Maxgrbv opened the pull request:
---
:mortar_board: [Сетка страниц и крупных блоков на флексах](https://up.htmlacademy.ru/htmlcss/35/user/2205519/tasks/10)

Maxgrbv committed:
Сетки страниц на флексах

senchkim commented on styles/styles.css:
```diff
@@ -100,46 +149,51 @@ body {
+  border-radius: 20px;
 }
 
 /* Подскажите, пожалуйста, почему я не могу стилизовать ссылку button-gift под кнопку? Ширина с высотой не изменяются */
```
Не совсем понимаю, зачем стилизовать ссылку как кнопку?

senchkim commented on styles/styles.css:
```diff
@@ -100,46 +149,51 @@ body {
   text-decoration: none;
-  display: none;   /*  Все равно не совсем понял, для чего она сейчас скрывается   */
+  display: none;
+  /*  Все равно не совсем понял, для чего она сейчас скрывается   */
```
о чем речь идет?)

senchkim commented on styles/styles.css:
```diff
@@ -148,38 +202,67 @@ border-radius: 26px;
+  padding: 0 40px;
+  width: 190px;
+  height: 44px;
+  /*    Пока не понимаю, как сделать корректный перенос на следующую строку     */
```
1) можешь через `<br />` - прямо в верстке
2) либо по играйся через `width`

senchkim commented on styles/styles.css:
```diff
@@ -189,32 +272,50 @@ border-radius: 26px;
 }
 
 .about-us-description {
+  float: right;
```
это надо УДАЛИТЬ!
никаких флоатов не должно быть

Сетку делаем через `flex` или `grid`

senchkim commented on styles/styles.css:
```diff
@@ -189,32 +272,50 @@ border-radius: 26px;
 
 .about-us-description {
+  float: right;
+  gap: 16px;  /*     gap не работает, не нонимаю как сделать отступ от svg иконок      */
```
Потому что и иконка это СТРОЧНЫЙ элемент, а строчные элементы не реагируют на отступы влево и вправо!

прочитай что такое строчный и блочные элементы и в чем разница

senchkim commented on styles/styles.css:
```diff
@@ -189,32 +272,50 @@ border-radius: 26px;
+}
+
+.about-img {
+  float: left;
```
удалить

senchkim commented on styles/styles.css:
```diff
@@ -247,21 +351,35 @@ border-radius: 26px;
+}
+
+.newsletter-field {
+padding: 16px 14px;
```
форматирование поехало

senchkim commented on styles/styles.css:
```diff
@@ -330,23 +474,109 @@ border-radius: 26px;
+.socials-footer-item {
+  list-style: none;
+  display: flex;
+  /* Почему иконки соц. сетей не выстраиваются в ряд после применения display: flex? */
```
Потому что нужно родителю задать `flex` а родитель в данном случае вот этот класс `socials-footer-list`

Maxgrbv committed:
Исправление флексов

> Не совсем понимаю, зачем стилизовать ссылку как кнопку?

Я имел в виду, что на макете самой первой кнопкой идёт кнопка "Заказать". Я так понимаю, что это настоящая кнопка а не ссылка, для неё нужна стилизация.

Следующими в макете, в секции с подарками, идут 2 белые ссылки (вроде как они должны вести на страницу с каталогом), которые выглядят как кнопки. Единственное, что у них отличается, как я вижу, это ширина из-за текста, и мы их почему-то скрыли, не смотря на то, что они есть в макете)
